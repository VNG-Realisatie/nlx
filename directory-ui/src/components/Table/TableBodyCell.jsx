import { node, oneOf } from 'prop-types'
import styled from 'styled-components'

const TableBodyCell = styled.td`
    display: table-cell;
    padding: .65rem;
    border: 2px solid transparent;

    font-size: 1rem;
    line-height: 1.9rem;
    font-weight: 400;
    color: #2D3240;
    
    text-align: ${ p => p.align};

    &:last-child {
        padding-right: 1.5rem;
    }
`

TableBodyCell.propTypes = {
  children: node,
  align: oneOf(['left', 'center', 'right']).isRequired,
}

TableBodyCell.defaultProps = {
  align: 'left',
}

export default TableBodyCell
