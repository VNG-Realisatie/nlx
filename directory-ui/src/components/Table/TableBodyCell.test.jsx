import React from 'react'
import { shallow } from 'enzyme'
import TableBodyCell from './TableBodyCell'

it('should match snapshot', () => {
  expect(shallow(<TableBodyCell/>)).toMatchSnapshot()
})

it('should render child elements', () => {
  expect(shallow(<TableBodyCell>
    <tr>
      <td>Table body</td>
    </tr>
  </TableBodyCell>).contains(<tr>
    <td>Table body</td>
  </tr>)).toEqual(true)
})
