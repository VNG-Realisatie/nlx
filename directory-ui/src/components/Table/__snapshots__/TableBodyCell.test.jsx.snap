// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<StyledComponent
  align="left"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    display: table-cell;
    padding: .65rem;
    border: 2px solid transparent;

    font-size: 1rem;
    line-height: 1.9rem;
    font-weight: normal;
    color: #424242;
    
    text-align: ",
          [Function],
          ";

    &:last-child {
        padding-right: 1.5rem;
    }
",
        ],
      },
      "defaultProps": Object {
        "align": "left",
      },
      "displayName": "styled.td",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "align": [Function],
        "children": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "td",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
